_source/_posts/2022-01-13-mobile-sso.md

```diff
@@ -0,0 +1,173 @@
+---
+layout: blog_post
+title: How to Create a Seamless Mobile SSO (Single Sign-On) Experience in iOS
+author: huan-liu
+by: internal-contributor
+communities: [mobile]
+description: "learn platform constraint on iOS platform and how to build great SSO experience around those constraints"
+tags: [ios, sso, mobile-sso, mobile]
+tweets:
+- ""
+- ""
+image: blog/mobile-sso/mobile-sso-social.jpeg
+type: conversion
+---
+
+On an iPhone, when we log in to an app, we click a login button, and a website pops up to verify our credentials. Once verified, the website then redirects back to the app, and you are logged in. This familiar Single Sign-On (SSO) pattern is frequently referred to as the *redirect flow* for authentication. The use of a web browser for auth in this example was considered [a "Best Current Practice"](https://www.rfc-editor.org/rfc/rfc8252.txt) for security and usability reasons.
+
+However, in 2017, a new prompt appeared in the login flow, before you were taken to the website. The following screenshot came from an iOS app preparing to log you in with Facebook. The interface prompt informs you that a specific app, the Yelp app in this example, wants to use a specific website to sign you in, and it warns you about information sharing. 
+
+{% img blog/mobile-sso/ASWebAuthenticationSession-prompt-from-yelp.jpeg alt:"app login prompt when authenticating with ASWebAuthenticationSession" width:"300" %}{: .center-image }
+
+There are a couple of problems with this prompt:
+
+* **Problem 1** (Where did this prompt come from?) : Many people consider this a bad user experience because the prompt looks out of place, and the wording is confusing and might alarm end users. 
```

Comment:
```suggestion
* **Problem One** (Where did this prompt come from?) : Many people consider this a bad user experience because the prompt looks out of place, and the wording is confusing and might alarm end users. 
```
style guide best practice